### ISSUE TITLE
REST reporter omits multipart requests.

### ISSUE THREAD

**Issue by AmyKosinova** [EmbedITCZ/jbehave-support#61]
![image](https://user-images.githubusercontent.com/19281259/51250552-01a72c80-1997-11e9-8c66-d6703c953000.png)

**Comment by paveljandejsek** [comment]
Can you post the example call of the rest step to reproduce?

**Comment by AmyKosinova** [comment]
When [POST] request to [APPLICATION]/[controller] is sent with data:
| name    |  data|
| @header.Content-Type | multipart/form-data |
| ticketCode           | {CP:TICKET_CODE}    |
| file                 | {RESOURCE:test.pdf} |

**Comment by paveljandejsek** [comment]
TLDR/ADHD version: **I dont know how to parse multipart/form-data, please help 🙁** 

This is going to be a tricky one... Main cause is this line of code in RestXmlReporterExtension:
```
private static MimeType[] compatibleMimeTypes =
        new MimeType[]{new MimeType("text"), new MimeType("application", "json"), new MimeType("application", "xml")};
```

But even if we would add multipart/form-data to it there is a problem - what to put into the report?
One way would be to put the whole body as is into report - user would see what was sent, but in the case of binary data of several MBs the report would not be readable (and would be too big)

Example of the String value we have:
`When [POST] request to [TEST]/[base64/multipart/] is sent with data:
| name                 | data                |
| @header.Content-Type | multipart/form-data |
| data                 | {RESOURCE:image.png}    |
| firstName            | someName            |
| lastName             | lastName            |`

```--4YNEx_sFwuIHyjppD_xUwUg71irYMGAqYN-w8hj4
Content-Disposition: form-data; name="data"; filename="image.png"
Content-Type: application/octet-stream
Content-Length: 34016

�PNG

   
IHDR       �ˬ�    cHRM  z&  ��  �   ��  u0  �`  :�  p��Q<   bKGD � � �����   	pHYs  ��  ��㑤"  � IDATx���w�U�����~���}�)�^H%!A:H,4;��
"�(���'��"����E)

looooooooooooooooooooooooooooots of data


a     �!,      6�     ���      �      �     CX      lܒ�     0/�?-DD��    IEND�B�
--4YNEx_sFwuIHyjppD_xUwUg71irYMGAqYN-w8hj4
Content-Disposition: form-data; name="firstName"
Content-Type: text/plain;charset=UTF-8
Content-Length: 8

someName
--4YNEx_sFwuIHyjppD_xUwUg71irYMGAqYN-w8hj4
Content-Disposition: form-data; name="lastName"
Content-Type: text/plain;charset=UTF-8
Content-Length: 8

lastName
--4YNEx_sFwuIHyjppD_xUwUg71irYMGAqYN-w8hj4--
```

We have the whole body that we received as a String that we could somehow parse to differentiate between parameters - tricky part, more or less only somehow reasonable solution would be something like [this](https://stackoverflow.com/questions/48410691/parse-a-string-containing-multipart-form-data-request-body-in-java)
At that point we could probably somehow put the 'text/plain' parameters into the report, but what should we do with the octet stream for files? Write something like _file: Binary Content_? Or try to save the binary and put it as an attachment somehow? (would be ideal I think)

Easiest solution (and probably most reasonable but not user friendly) I have so far is for multipart data request to at least write something in the report like: _multipart/form-data_ and put a link to external txt.file that would contain the whole String body that we save into it.

I would very much like to see this one solved somehow, but right now neither of the solutions seems ideal - anyone has any advice?
